readthedocs/oauth/utils.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+from django.contrib import messages
+from django.utils.translation import ugettext_lazy as _
+
+from readthedocs.oauth.services import registry
+
+log = logging.getLogger(__name__)
+
+
+def attach_webhook(project, request=None):
+    """Add post-commit hook on project import"""
+
+    for service_cls in registry:
+        if service_cls.is_project_service(project):
+            service = service_cls
+            break
+    else:
+        return None
+
+    user_accounts = service.for_user(request.user)
+    for account in user_accounts:
+        success, resp = account.setup_webhook(project)
```

Comment:
It was useful for debugging, and I think it will let us give the user more information in errors in the future. There are more states than "True or False" I think around this crazy webhook logic. We could possibly setup a few more states in a constants for 400/404/500 etc.